### DISCUSSION TITLE
Best Synology NAS for frigate

### DISCUSSION THREAD

**Discussion by Solarflor** [blakeblackshear/frigate#7127] [General]
Hi all,
I'm currently using frigate as integration on HASSIO on my RPi 4 but I'm thinking to move on a NAS.
Not sure if to install frigate standalone or as addon of HASSIO but my question is which NAS I should take.

I have to decided between Synology DS720+ or DS723+
My (poor) undertsanding is that the 723+ is a newer version but without GPO and frigate needs to work on GPO so ....
... I need your support.
Thanks

**Comment by doctorttt** [comment]
If you were to go with Synology DS+ NAS, it would be much easier to just use Synology's Surveillance Station. Those two Synology models are too weak to run docker containers.

**Reply by Solarflor** [reply]
Thanks doctorttt,
the fact is that the cost of the additional cards to add more cameras I could almost buy another NAS.
By the way which Synology NAS are you suggesting that could be suitable for the scope?
thanks again

**Reply by microraptor** [reply]
I think doctorttt mixed up the model numbers. DS720+ and DS723+ are powerful enough to run all kinds of docker containers. The official hardware recommendation site of Frigate only suggests three computers, of which two run the Intel Celeron J4125 with its integrated graphics. As a matter of fact the DS720+ has the exact same CPU/GPU. In stock configuration the RAM is a little small, so I recommend to add at least a 4GB module ([which isn't very expensive](https://nascompares.com/2022/06/01/synology-unofficial-memory-testing-on-dsm-7-1-ds920-ds220-ds720-ds420-and-ds1520-nas/)).

You are correct about the DS723+ not having a GPU, which can be used by Frigate for some hardware transcoding stuff. However, the Ryzen CPU is much more powerful than the Celeron and can do the transcoding in software instead. This is less energy efficient though and probably a lot slower, but I haven't found any transcoding benchmarks with the two. So it is probably better to get a Synology with the Celeron instead of the newer Ryzen, if its main purpose is Frigate or Plex/Jellyfin.

Small Synology NAS with a Intel Celeron J4125:
- DS224+ (2-bay)
- DS720+ (2-bay)
- DS423+ (4-bay)
- DS920+ (4-bay)

Small Synology NAS with a AMD Ryzen R1600 and no GPU (as of 2023):
- DS723+ (2-bay)
- DS923+ (4-bay)
- DS1522+ (5-bay)

**Reply by Solarflor** [reply]
Very Clear.
Thanks

**Reply by microraptor** [reply]
Sorry, I got the list wrong. DS423+ is 4-bay and I forgot about DS224+. The list is fixed now.

**Reply by Solarflor** [reply]
> I forgot about DS224+

My understanding is that there is no reason to choose DS224+ if compared with the old DS220+.
But probably it is to small for frigate?

**Reply by microraptor** [reply]
> My understanding is that there is no reason to choose DS224+ if compared with the old DS220+. But probably it is to small for frigate?

Comparing the specs, the only difference seems to be that DS220+ runs the less powerful Intel Celeron J4025, instead of the J4125 of the DS224+. So I recommend the DS224+, unless there is a huge difference in pricing.

I am also trying to find a new Synology for family. So far these are my notes to the small models, assuming pricing is reasonable and budget isn't a limiting factor:

If a GPU is not required, choose one with a Ryzen, otherwise the DS423+ seems to me to be the best followed by the DS920+. If you are really sure you won't need more than two bays future, the DS720+ is also a good choice. These three models have two M.2 slots for SSDs (besides the 2 or 4 SATA bay slots). I would add extra (unofficial) RAM regardless of model.

DS423+ compared to DS920+: 
- No eSata for DX517 unit expansion
- 2GB less built-in RAM
- Official storage volume support for the M.2 slots with Synology's  SSDs (both have the slots for caching)

DS920+ compared to DS720+:
- Two more SATA bays; bigger size
- 2GB more built-in RAM
- Software according to spec sheet: some backup software; object recognizion in synology photos

DS720+ compared to DS224+/DS220+:
- Has 2x M.2 SSD slots (officially only for caching)
- Has eSata for DX517 unit expansion
- Software according to spec sheet: VM virtualization
- Missing software according to spec sheet (DS224+/DS220+ have it listed): some backup softeware; object recognizion in synology photos

About the M.2 SSD slots: 
I plan on putting two third-party NVMe SSDs into the M.2 slot. Only the DS423+ and the Ryzen models support using the M.2 slots for storage volumes, but only with ridiculously expensive (and low capacity) Synology SSDs. However, [there is this script](https://github.com/007revad/Synology_HDD_db/), which allows to add any SSD to the compatibility list and then use them as storage in Synology's software. Although on the DS920+ and DS720+ the M.2 SSDs are officially only supported for caching, it is possible to format them as a storage volume [with this script](https://github.com/007revad/Synology_M2_volume) or [some CLI commands](https://old.reddit.com/r/synology/comments/pwrch3/how_to_create_a_usable_poolvolume_to_use_as/). Either way the Synology_HDD_db script should be run after every update to mitigate any issues. [See here for more information about the different scripts](https://github.com/007revad/Synology_HDD_db/discussions/81#discussioncomment-5984808). I think the DS423+ is the best choice, because it has better native NVMe storage support.

Spec sheets:
https://www.synology.com/en-us/support/download/DS423+?version=7.2#docs
https://www.synology.com/en-us/support/download/DS920+?version=7.2#docs
https://www.synology.com/en-us/support/download/DS720+?version=7.2#docs
https://www.synology.com/en-us/support/download/DS224+?version=7.2#docs
https://www.synology.com/en-global/compatibility?search_by=products&model=DS923%2B&category=m2_ssd_internal

**Reply by Solarflor** [reply]
👍

**Reply by Solarflor** [reply]
At the end I bought a DS720+ with 8GB additional RAM.
I installed frigate in Container and added 3 cameras,
I have a question about CPU utilization.
Looking to the CPU utilization within Container Manager the utilization is around 30% that is not too high but a bit worring considering I would like to add more cameras.
![image](https://github.com/blakeblackshear/frigate/assets/100423265/9e861982-c4e4-4229-bbb0-93684d6f2a6d)
but lookig to the CPU utilization in Control Panel in around 10%, much better.
![image](https://github.com/blakeblackshear/frigate/assets/100423265/27e02cdf-36bb-480c-9849-c267a1ba16c1)


Which one I should consider?

**Reply by microraptor** [reply]
I actually haven't set up a Synology yet, so I am not familiar with the UI and why the CPU utilization is different. Perhaps the container has a resource limit and cannot use all cores? The CPU utilization in the first screenshot is also close to the load average. So perhaps it is mislabeled and actually the linux load average, which can go beyond 100%.

I am surprised only three cameras take up so much resources.  Are you using [cameras with multiple substreams in different resolutions](https://docs.frigate.video/frigate/hardware/#cameras) or are re-encoding the same stream for different tasks?

**Reply by Solarflor** [reply]
I'm using dahua cameras with the following setting
![image](https://github.com/blakeblackshear/frigate/assets/100423265/bda19755-0a8e-4670-bd4f-386177bea898)

This is the config used for each camera

```
mqtt:
 host: 192.168.1.13 #core-mosquitto
 port: 1883
 user: xxxxx
 password: xxxxx

# Optional: birdseye configuration
# NOTE: Can (enabled, mode) be overridden at the camera level
birdseye:
  # Optional: Enable birdseye view (default: shown below)
  enabled: True
  # Optional: Restream birdseye via RTSP (default: shown below)
  # NOTE: Enabling this will set birdseye to run 24/7 which may increase CPU usage somewhat.
  restream: False
  # Optional: Width of the output resolution (default: shown below)
  width: 1280
  # Optional: Height of the output resolution (default: shown below)
  height: 720
  # Optional: Encoding quality of the mpeg1 feed (default: shown below)
  # 1 is the highest quality, and 31 is the lowest. Lower quality feeds utilize less CPU resources.
  quality: 5
  # Optional: Mode of the view. Available options are: objects, motion, and continuous
  #   objects - cameras are included if they have had a tracked object within the last 30 seconds
  #   motion - cameras are included if motion was detected in the last 30 seconds
  #   continuous - all cameras are included always
  mode: objects


#ffmpeg:
#  hwaccel_args: preset-rpi-64-h264
detect:
  enabled: True # <---- disable detection until you have a working camera feed
  width: 1920 # <---- update for your camera's resolution
  height: 1080 # <---- update for your camera's resolution
motion:
record:
  # Optional: Enable recording (default: shown below)
  # WARNING: If recording is disabled in the config, turning it on via
  #          the UI or MQTT later will have no effect.
  enabled: true
  # Optional: Number of minutes to wait between cleanup runs (default: shown below)
  # This can be used to reduce the frequency of deleting recording segments from disk if you want to minimize i/o
  expire_interval: 60
  # Optional: Retention settings for recording
  retain:
    # Optional: Number of days to retain recordings regardless of events (default: shown below)
    # NOTE: This should be set to 0 and retention should be defined in events section below
    #       if you only want to retain recordings of events.
    days: 0
    # Optional: Mode for retention. Available options are: all, motion, and active_objects
    #   all - save all recording segments regardless of activity
    #   motion - save all recordings segments with any detected motion
    #   active_objects - save all recording segments with active/moving objects
    # NOTE: this mode only applies when the days setting above is greater than 0
    mode: all
  # Optional: Event recording settings
  events:
    # Optional: Number of seconds before the event to include (default: shown below)
    pre_capture: 3
    # Optional: Number of seconds after the event to include (default: shown below)
    post_capture: 3
    # Optional: Objects to save recordings for. (default: all tracked objects)
    objects:
      - person
    # Optional: Restrict recordings to objects that entered any of the listed zones (default: no required zones)
    required_zones: []
    # Optional: Retention settings for recordings of events
    retain:
      # Required: Default retention days (default: shown below)
      default: 10
      # Optional: Mode for retention. (default: shown below)
      #   all - save all recording segments for events regardless of activity
      #   motion - save all recordings segments for events with any detected motion
      #   active_objects - save all recording segments for event with active/moving objects
      #
      # NOTE: If the retain mode for the camera is more restrictive than the mode configured
      #       here, the segments will already be gone by the time this mode is applied.
      #       For example, if the camera retain mode is "motion", the segments without motion are
      #       never stored, so setting the mode to "all" here won't bring them back.
      mode: motion
      # Optional: Per object retention days
      objects:
        person: 5
        car: 2
        truck: 2
        bicycle: 2
        motorcycle: 2
        dog: 2
        cat: 2


cameras:
  Piazzale_1: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://user:passwd@192.168.1.209:554/cam/realmonitor?channel=1&subtype=1 # <----- The stream you want to use for detection
          roles:
            - detect
            - rtmp
    objects:
      track:
        - person
    record: # <----- Enable recording
      enabled: True
    snapshots: # <----- Enable snapshots
      enabled: True
    zones:
      zone_1:
        coordinates: 602,230,1847,551,1654,1080,0,1080
  Ingresso: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://user:passwd@192.168.1.211:554/cam/realmonitor?channel=1&subtype=1 # <----- The stream you want to use for detection
          roles:
            - detect
            - rtmp
    objects:
      track:
        - person
    record: # <----- Enable recording
      enabled: True
    snapshots: # <----- Enable snapshots
      enabled: True
    zones:
      zone_1:
        coordinates: 1892,1048,1305,0,681,0,427,1080
  Cancello: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://user:passwd@192.168.1.208:554/cam/realmonitor?channel=1&subtype=1 # <----- The stream you want to use for detection
          roles:
            - detect
            - rtmp
    objects:
      track:
        - person
    record: # <----- Enable recording
      enabled: True
    snapshots: # <----- Enable snapshots
      enabled: True
    zones:
      zone_1:
        coordinates: 1899,1037,987,242,0,309,0,1080
```

**Reply by NickM-27** [reply]
So you're recording the sub stream and telling Frigate to scale the 704x480 stream up to 1920x1080 for detect

That is very inefficient

**Reply by Solarflor** [reply]
I'm not surprised knowing what is my knowledge on this matter.
Could you please kindly suggest an optimal configuration?
Thanks

**Reply by NickM-27** [reply]
1. I'd suggest using https://docs.frigate.video/guides/getting_started#step-6-enable-recording-andor-snapshots so you can record the high res stream. 
2. From your screenshot it looks like your camera supports a second stream. If so I'd recommend setting that to 1280x720 and using rtsp://user:passwd@192.168.1.209:554/cam/realmonitor?channel=1&subtype=2 along with setting the detect resolution at the camera level

```
cameras:
  Ingresso:
    detect:
      width: 1280
      height: 720
```

**Reply by Solarflor** [reply]
Many thanks.
I will try it and let you know 👍

**Comment by Solarflor** [comment]
Just to be sure is this what you are suggesting?
stream 1 for recording and stream 2 for detect?
![image](https://github.com/blakeblackshear/frigate/assets/100423265/4a840d0c-fa27-47fd-b837-eceec35dbeef)

**Reply by NickM-27** [reply]
I use subtype=0 (2688x1520) for record and subtype=2 (1280x720) for detect

**Comment by Solarflor** [comment]
ok, I will try

**Comment by Solarflor** [comment]
Hi guys,
I need another help from you.
I'm not able to find the right input for the following camera

Camera REIGY
Device Name W-NVR
Unit Type K8210-3WS
Equipment Sequence Number 9021160200
Software Version 3.2.4.9M
Modello IP CAMERA: XF-A2628D-B

In home hassistant  I was able to see it as "Generic Camera" with the followinig url
http://admin:password@192.168.1.17/snapshot.jpg?user=admin&pwd=admin&strm=6

I have tried different type of rtsp but without success.

Following the documentation I have tried
```
go2rtc:
  streams:
    mjpeg_cam: "ffmpeg:http://admin:password@192.168.1.17/snapshot.jpg?user=admin&pwd=&strm=6#video=h264#hardware" 
cameras:
  mjpeg_cam:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/mjpeg_cam
          input_args: preset-http-jpeg-generic
          roles:
            - detect
            - record
```
but no success again
Thanks
